lib/solar.js

```diff
@@ -1,175 +1,232 @@
+function _inverse_longitude(l) {
   // Make an initial guess based on the mean longitude.
   let t = (l - Q0) / Q1;
 
   // Iteratively improve the guess using Newton's method. (I've never seen
   // any significant accuracy improvement past two iterations, so I use
   // three, just to be safe. :) )
-  t = t - (longitude(t) - l) / longitude_prime(t);
-  t = t - (longitude(t) - l) / longitude_prime(t);
-  t = t - (longitude(t) - l) / longitude_prime(t);
+  t -= (_longitude(t) - l) / _longitude_prime(t);
+  t -= (_longitude(t) - l) / _longitude_prime(t);
+  t -= (_longitude(t) - l) / _longitude_prime(t);
 
   return t;
 }
 
-// Convert Julian date t to the obliquity of the ecliptic
-function obliquity(t) {
+// obliquity of the ecliptic
+const E0 = 23.439 * RAD;
+const E1 = -0.00000036 * RAD;
+function _obliquity(t) {
   return E0 + E1 * t;
 }
 
-// Convert Julian date t to the equatorial right ascension
-function right_ascension(t) {
-  const l = longitude(t);
-  return Math.atan2(Math.sin(l) * Math.cos(obliquity(t)), Math.cos(l));
-}
-
-// Convert Julian date t to the equatorial declination
-function declination(t) {
-  return Math.asin(Math.sin(longitude(t)) * Math.sin(obliquity(t)));
-}
-
+// distance to the sun in AU
 const D0 = 1.00014;
 const D1 = -0.01671;
 const D2 = -0.00014;
-
-// Convert Julian date t to the distance between the sun and the Earth-Moon
-// barycenter in AU
-function distance(t) {
-  const g = mean_anomaly(t);
+function _distance(t) {
+  const g = _mean_anomaly(t);
   const cos_g = Math.cos(g);
   const cos_2g = 2 * cos_g * cos_g - 1;
   return D0 + D1 * cos_g + D2 * cos_2g;
 }
 
-// Transit time constants.
+// solar transit time
 const J0 = 0.0009; // Leap second correction
+function _julian_cycle(t, lon) {
+  return Math.round(t - J0 + lon * (0.5 / Math.PI));
+}
+
+// https://en.wikipedia.org/wiki/Sunrise_equation#Mean_solar_noon
+function _mean_solar_noon(t, lon) {
+  return J0 - lon * (0.5 / Math.PI) + _julian_cycle(t, lon);
+}
+
+// https://en.wikipedia.org/wiki/Sunrise_equation#Solar_transit
 const T0 = 0.0053;
 const T1 = -0.0069;
+function _transit(t, lon) {
+  const t0 = _mean_solar_noon(t, lon);
+  return t0 + T0 * Math.sin(_mean_anomaly(t0)) + T1 * Math.sin(2 * _longitude(t0));
+}
+
+function _altitude(sin_h0, rise, t, lat, lon) {
+  const sin_dec = Math.sin(_longitude(t)) * Math.sin(_obliquity(t));
+  const cos_dec = Math.sqrt(1 - sin_dec * sin_dec);
 
-function julian_cycle(t, lon) {
-  return Math.round(t - J0 + lon/(2 * Math.PI));
+  return _transit(t, lon) + rise * Math.acos(
+    (sin_h0 - Math.sin(lat) * sin_dec) / (Math.cos(lat) * cos_dec)
+  );
 }
 
-// https://en.wikipedia.org/wiki/Sunrise_equation#Mean_solar_noon
-function mean_solar_noon(t, lon) {
-  return J0 - lon / (2 * Math.PI) + julian_cycle(t, lon);
+
+// Display methods. Internally we work with Julian dates and radians.
+// Externally we work with millisecond timestamps and degrees.
+
+function longitude(ms) {
```

Comment:
These looked so similar to `lunar.js` that i was going to offer to factor them out (before i realized they're totally different 🤣 ) you made the modules too good 😉